### PR TITLE
GCS fixes

### PR DESCRIPTION
**Pull request by zimnx**

### CODE REVIEW

**Comment by mmatczuk** [comment]
Please change replace import path to scylla repo upon merging PR.

This https://github.com/scylladb/rclone/pull/7/commits/9cbc38a484c61c2cac3ccc70fca717766e9a102a should be sent to rclone upstream.

File: backend/googlecloudstorage/googlecloudstorage.go [hunk 1]
```diff
@@ -1063,9 +1113,21 @@ func (o *Object) Update(ctx context.Context, in io.Reader, src fs.ObjectInfo, op
...
+
+	// Google library expects buffer of 0 len with enough capacity.
+	buf = buf[:0]
+	defer o.fs.pool.Put(buf)
+
 	var newObject *storage.Object
 	err = o.fs.pacer.CallNoRetry(func() (bool, error) {
-		insertObject := o.fs.svc.Objects.Insert(bucket, &object).Media(in, googleapi.ContentType("")).Name(object.Name)
+		insertObject := o.fs.svc.Objects.Insert(bucket, &object).
+			Media(in,
```
Comment (mmatczuk) [inline backend/googlecloudstorage/googlecloudstorage.go]: nit: formatting could be improved

File: backend/googlecloudstorage/googlecloudstorage.go [hunk 2]
```diff
@@ -430,6 +474,12 @@ func NewFs(name, root string, m configmap.Mapper) (fs.Fs, error) {
 		opt:   *opt,
 		pacer: fs.NewPacer(pacer.NewGoogleDrive(pacer.MinSleep(minSleep))),
 		cache: bucket.NewCache(),
+		pool: pool.New(
+			time.Duration(opt.MemoryPoolFlushTime),
+			int(opt.ChunkSize),
+			fs.Config.Transfers,
```
Comment (mmatczuk) [inline backend/googlecloudstorage/googlecloudstorage.go:486]: In S3 we use `opt.UploadConcurrency*fs.Config.Transfers,` which makes sense for a pool `poolSize`?

File: backend/googlecloudstorage/googlecloudstorage.go [hunk 3]
```diff
@@ -1063,9 +1113,21 @@ func (o *Object) Update(ctx context.Context, in io.Reader, src fs.ObjectInfo, op
 		ContentType: fs.MimeType(ctx, src),
 		Metadata:    metadataFromModTime(modTime),
 	}
+
+	buf := o.fs.pool.Get()
+
+	// Google library expects buffer of 0 len with enough capacity.
+	buf = buf[:0]
```
Comment (mmatczuk) [inline backend/googlecloudstorage/googlecloudstorage.go]: This is a good practice I think that this is handled by pool impl, if it isn't it should be.

File: backend/googlecloudstorage/googlecloudstorage.go [hunk 4]
```diff
@@ -1075,6 +1137,7 @@ func (o *Object) Update(ctx context.Context, in io.Reader, src fs.ObjectInfo, op
 	if err != nil {
 		return err
 	}
+
```
Comment (mmatczuk) [inline backend/googlecloudstorage/googlecloudstorage.go]: Not related to the patch.

File: backend/googlecloudstorage/googlecloudstorage.go (hunk 2 above)
Comment (zimnx) [reply backend/googlecloudstorage/googlecloudstorage.go:486]: GCS doesn't have `UploadConcurrency`, there are only `fs.Config.Transfers` in parallel.

**Comment by zimnx** [comment]
> Please change replace import path to scylla repo upon merging PR.
> 

Changed replace. 

> This [9cbc38a](https://github.com/scylladb/rclone/commit/9cbc38a484c61c2cac3ccc70fca717766e9a102a) should be sent to rclone upstream.

I'll wait couple of days, maybe patch to google lib will be merged and released. If so I'll submit all the changes.

**Pull request merged**